File: common/src/test/java/munoon/bank/common/validation/money/MoneyCountValidatorTest.java
package munoon.bank.common.validation.money;

import lombok.AllArgsConstructor;
import org.junit.jupiter.api.Test;

import javax.validation.Validation;
import javax.validation.Validator;

import static org.assertj.core.api.Assertions.assertThat;

class MoneyCountValidatorTest {
    private final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();

    @Test
    void testValid() {
        TestData testData = new TestData(1.0);
        var result = validator.validate(testData);
        assertThat(result).isEmpty();
    }

    @Test
    void testInvalid() {
        TestData testData = new TestData(1.00001);
        var result = validator.validate(testData);
        assertThat(result).isNotEmpty();
    }

    @Test
    void testNull() {
        TestData testData = new TestData(null);
        var result = validator.validate(testData);
        assertThat(result).isEmpty();
    }

    @AllArgsConstructor
    private static class TestData {
        @ValidMoneyCount
        Double value;
    }
}

File: transactional-service/src/test/kotlin/munoon/bank/service/transactional/util/validator/MoneyCountValidatorTest.kt
package munoon.bank.service.transactional.util.validator

import munoon.bank.service.transactional.AbstractTest
import munoon.bank.service.transactional.util.validation.CardType
import munoon.bank.service.transactional.util.validation.ValidMoneyCount
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import org.springframework.beans.factory.annotation.Autowired
import javax.validation.Validator

internal class MoneyCountValidatorTest : AbstractTest() {
    @Autowired
    private lateinit var validator: Validator

    @Test
    fun isValid() {
        val data = TestData(1.0)
        val result = validator.validate(data)
        assertThat(result).hasSize(0)
    }

    @Test
    fun testInvalid() {
        val data = TestData(1.00001)
        val result = validator.validate(data)
        assertThat(result).hasSize(1)
    }

    @Test
    fun testNull() {
        val data = TestData(null)
        val result = validator.validate(data)
        assertThat(result).hasSize(0)
    }

    private data class TestData(
            @field:ValidMoneyCount
            val count: Double?
    )
}